### ISSUE TITLE
[no-unused-expressions] false positive on dynamic import()

### ISSUE THREAD

**Issue by derekparsons718** [typescript-eslint/typescript-eslint#2108]
I believe I'm getting a false positive for `no-unused-expressions` when I use dynamic import statements.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-expressions": "error"
  }
}
```

```TS
//At the top of a file
import("./app/app.module"); //no-unused-expressions error
```

**Expected Result**

No linting error. The dynamic import is not unused or unnecessary.

**Actual Result**

The line is marked as a "no-unused-expressions" error.

**Additional Info**

Adding a `.then()` removes the error.

```TS
import("./app/app.module").then(); //no error
```

This error only appeared after I upgraded from v2.34.0 to v3.0.1. I didn't see anything in the release notes about this rule changing.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.1` |
| `@typescript-eslint/parser`        | `3.0.1` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `13.11.0` |
| `npm`                              | `6.13.7` |

**Comment by bradzacher** [comment]
yeah this is just not handled.

OOC - how often do you have a dangling `import('foo')`? They're async imports so it seems weird to have one with no `await` or `.then`.  
Or was this just something you ran into randomly whilst typing?

(note - adding an `await` or a `.then` fixes it because it converts it into an expression the base rule understands)

**Comment by derekparsons718** [comment]
We use dynamic imports to accomplish [code-splitting in webpack](https://webpack.js.org/guides/code-splitting/#dynamic-imports) and to enable [prefetching/preloading via webpack](https://webpack.js.org/guides/code-splitting/#prefetchingpreloading-modules).

To my understanding, the example I gave doesn't need a `.then` because it is loading an angularjs application, which I guess ends up bootstrapping itself as a sort of side-effect. Another place we use a dangling import() is to load css files via webpack, which we could do with a static import but we wanted to take advantage of the code-splitting and prefetching that comes with the dynamic imports. Obviously, a css file loaded via webpack doesn't need a `.then` either.

To be honest, I'm not an expert in webpack so maybe these are bad reasons haha, but hopefully that answers your question.

**Comment by bradzacher** [comment]
That makes sense yeah.

It wouldn't be a difficult fix if you'd like to give it a go!
I believe all you need to check for and ignore is an `ImportExpression`.

**Comment by derekparsons718** [comment]
I've never contributed to an open source project before, but I'd love to take a crack at it.

Unfortunately I won't be able to start for a couple weeks, but if the issue is still open at that point I'll definitely come back to it.
